File: Resource/51900286-51900712-51900718-51900699/source demo/Web Application/ComNha_API/ComNha_API/Models/Nhacungcap.cs
using System;
using System.Collections.Generic;

namespace ComNha_API.Models
{
    public partial class Nhacungcap
    {
        public Nhacungcap()
        {
            Phieunhaps = new HashSet<Phieunhap>();
        }

        public string MaNhaCungCap { get; set; } = null!;
        public string MoTaNhaCungCap { get; set; } = null!;
        public string TenNhaCungCap { get; set; } = null!;
        public string DiaChi { get; set; } = null!;
        public string SoDienThoai { get; set; } = null!;

        public virtual ICollection<Phieunhap> Phieunhaps { get; set; }
    }
}

File: Dev/51900718/src/main/java/tdtu/advanced/java/thinh68/models/NhaCungCap.java
package tdtu.advanced.java.thinh68.models;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Table(name = "nha_cung_cap")
public class NhaCungCap {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ma_nha_cung_cap")
	private Long maNhaCungCap;

	@Column(name = "mo_ta_nha_cung_cap")
	private String moTaNhaCungCap;

	@Column(name = "ten_nha_cung_cap")
	private String tenNhaCungCap;

	@Column(name = "dia_chi")
	private String diaChi;

	@Column(name = "so_dien_thoai", unique = true, nullable = false)
	private String soDienThoai;

	@OneToMany(mappedBy = "nhaCungCap")
	private List<PhieuNhap> nhaCungCap_PhieuNhaps;
}